frontend_tests/casper_tests/05-subscriptions.js

```diff
@@ -79,8 +75,29 @@ casper.waitForText('Already subscribed', function () {
     casper.fill('form#add_new_subscription', {stream_name: '  '});
     casper.click('form#add_new_subscription input.btn');
 });
-casper.waitForText('Error adding subscription', function () {
-    casper.test.assertTextExists('Error adding subscription', "Can't subscribe to an empty stream name");
+casper.waitForText('Create stream', function () {
+    casper.test.assertTextExists('Create stream', 'Modal for specifying new stream users');
+    casper.fill('form#stream_creation_form', {stream_name: '  '});
+    casper.click('form#stream_creation_form button.btn.btn-primary');
+});
+casper.then(function () {
+    casper.test.info('Streams should be filtered when typing in the create box');
+});
+casper.then(function () {
+    casper.test.assertSelectorHasText('.subscription_row .subscription_name', 'Verona', 'Verona stream exists before filtering');
+    casper.test.assertSelectorDoesntHaveText('.subscription_row.notdisplayed .subscription_name', 'Verona', 'Verona stream shown before filtering');
+});
```

Comment:
I would expect us to need a `casper.waitForSomething` type call here to make sure the browser has time to update based on the click a couple lines above.  Otherwise, this test will fail nondeterministically, based on whether the browser update or Casper doing the next step happens first.

Also, I think you can collapse the `casper.test.info` block with the `casper.test.assert...` ones here.